packages/lib/src/components/Klarna/components/KlarnaContainer/KlarnaContainer.tsx

```diff
@@ -1,21 +1,45 @@
 import { h } from 'preact';
+import { useEffect, useRef, useState } from 'preact/hooks';
 import { KlarnaWidget } from '../KlarnaWidget/KlarnaWidget';
-import { useState } from 'preact/hooks';
+import type { ComponentMethodsRef, PayButtonFunctionProps, UIElementStatus } from '../../../internal/UIElement/types';
+import type { ActionHandledReturnObject } from '../../../../types/global-types';
+import type { AdyenCheckoutError, KlarnaAction, KlarnaComponentRef } from '../../../../types';
 
-export function KlarnaContainer(props) {
-    const [action, setAction] = useState({
-        sdkData: props.sdkData,
-        paymentMethodType: props.paymentMethodType,
-        paymentData: props.paymentData
-    });
+interface KlarnaContainerProps {
+    setComponentRef: (ref: ComponentMethodsRef) => void;
+    displayName: string;
+    showPayButton: boolean;
+    type: string;
+    onComplete(state: any): void;
+    onError(error: AdyenCheckoutError): void;
+    payButton(props?: PayButtonFunctionProps): h.JSX.Element;
+    onLoaded(): void;
+    onActionHandled(actionHandled: ActionHandledReturnObject): void;
+}
+
+export function KlarnaContainer({ setComponentRef, ...props }: KlarnaContainerProps) {
+    const [widgetInitializationTime, setWidgetInitializationTime] = useState<number>(null);
+    const [action, setAction] = useState<KlarnaAction>();
     const [status, setStatus] = useState('ready');
+    const klarnaRef = useRef<KlarnaComponentRef>({
+        setAction: (action: KlarnaAction) => {
+            setAction(action);
+            setWidgetInitializationTime(new Date().getTime());
```

Comment:
We use the 'initialization time' to keep track when the widget needs to be reinitialized. Changing it will trigger the effect in the KlarnaWidget component.

So, to keep things consistent, I set it here so we have the initial time that it got initialized once we get the action (would be weird to have it `null` or `0`).

Reply:
It's an unnecessary line of code, that isn't required for the component to work correctly.
I think it makes the workings of the component seem more complex than they need to be. 
And if you look at the sequence of events: `reinitializeWidget` is always called _before_ `setAction` - so `widgetInitializationTime` is always set

Reply:
Ok. It wasn't actually doing anything before, but now the change is in for [setting active pm when needed](https://github.com/Adyen/adyen-web/pull/3007/commits/2066f71d0be90d4b1025260b727c1c288894b510) - it becomes the prime driver for setting `widgetInitializationTime` (at least in regards to issue 1)